sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -95,1056 +39,288 @@
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-
 
 /**
- * Utility implementing the necessary reflection for working with {@link DoFn}s.
+ * Parses a {@link DoFn} and computes its {@link DoFnSignature}. See {@link #getOrParseSignature}.
  */
 public abstract class DoFnReflector {
```

Comment:
I don't think this needs to be abstract anymore. Previously, instances of `DoFnReflector` were essentially `DoFnSignature`s.